src/push.rs

```diff
@@ -23,6 +24,10 @@ lazy_static! {
         .timeout(REQWEST_TIMEOUT_SEC)
         .build()
         .unwrap();
+    static ref ASYNC_HTTP_CLIENT: AsyncClient = AsyncClient::builder()
```

Comment:
In my eyes defining an HTTP client as a global static is problematic, for all the reasons global variables are. While that is somewhat subjective, creating both an async client and a sync client, even though a user likely only needs one, sounds even worse.

What do people think of requiring users to create a client on demand and binding `push` as a method to such client?

Reply:
I agree with you that the globals aren't ideal, but I feel like lazy_static is a very ergonomic solution for consumers of the library. I like the idea of the user instantiating a client, but it's got a few little usability woes:

- Users now have to explicitly depend on reqwest and familiarize themselves with it works, how to create a client, etc.
- It's a breaking change (I know this is still a 0.x crate, so breaking changes are expected, but we should try to minimize them)

What if instead, we just made async support a feature that users had to opt into? That way we could instantiate the async client if and only if the feature is enabled?

Reply:
I fear than adding an additional either-or feature is only making the whole crate harder to test and maintain.

Reply:
Agreed. I've removed the feature now.